proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,456 @@
+change for a feature requested by the community. It will also reduce the amount
+of context switching for the maintainers, allowing them to focus on giving high
+quality feedback to users. As we use this framework to process existing feature
+requests, we can revisit what the in flight limit should be.
+
+### Q. Why do we require issues to be opened instead of just pull requests?
+
+Having a defined process for intake reduces the burden on deciding which way is
+the best way. We're taking an opinionated stance that code changes made in
+isolation generally do not provide enough context on what problem exists and why
+the proposed change is the best way to fix it. Reviewing a pull request also
+requires significant effort on the part of maintainers - reading someone elses
+code can be an intensive task. Before dedicating that effort, we feel that
+having a more general discussion about the problem and solutions will help
+reduce the overall effort into code review.
+
+Pull requests are often made for problems that only affect the contributor or a
+very small portion of the user base. Requiring an issue provides a mechanism to
+request structured feedback in the form of "upvotes" or GitHub reactions to
+estimate the impact of the issue on the community.
+
+Logistically, using GitHub issues also centralizes the metadata on what work is
+available and ongoing via labels. While pull requests are really GitHub issues
+with code changes attached to them, it is difficult to interrogate issues and
+pull requests together to gather metrics, check on statuses and assignments, and
+perform automated tasks. Having all of the labels and data on a single entity
+type reduces the burden on maintaining the system.
+ 
+We do not intend this to be a gatekeeping or pointless task. If a user does open
+a pull request without an issue, we will use the opportunity to educate them
+about our process, and when necessary assist them in opening an issue. We can
+also open the issue ourselves on behalf of the user as well.
+
+### Q. Why would we re-prioritize contributions for review instead of taking them first in, first out?
+
+By default issues should be selected for review based on the stated criteria,
+and existing issues should not be pulled up above other existing issues.
+However, there are scenarios where a new issue may be added to the queue in an
+'insertion sort' fashion. One example considers if the issues currently
+prioritized for review are all large and complex. In this case, they risk
+blocking smaller and easier changes. We may pull up the smaller issues to keep
+the queue moving. In another case, a new feature may increase in urgency due to
+other API changes. Since the main responsibility of the AWS CLI is to facilitate
+access to the AWS API, we would re-prioritize a feature request to address that
+change.
+### Q: What do we do if a user requests a feature or change that would involve a major change in the way the SDK currently works?
```

Comment:
What part of the specification is this rationale question addressing? In general, I'm not sure how needed this question is as it relates to community contributions even after completion of the proposal it would be we would just not mark it as feasible to contribute because it is too complicated. Maybe we can add the proposal suggestion in a future works section?

Reply:
Referred to a section since removed. Agree it could go to future works.